bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java

```diff
@@ -59,7 +59,21 @@
  * @see <a href="http://www.eclipse.org/swt/examples.php">SWT Examples: GraphicsExample, PaintExample</a>
  * @see <a href="http://www.eclipse.org/swt/">Sample code and further information</a>
  */
-public final class GC extends Resource {
+public sealed class GC extends Resource {
+
+	public static final class Closeable extends GC implements AutoCloseable {
+		Closeable(Drawable drawable, int style) {
+			super(drawable, style);
+		}
+		@Override
+		public void close() {
+			dispose();
+		}
+	}
+
+	public static GC.Closeable create(Drawable drawable) {
```

Comment:
The new method needs some java-doc on all platforms. The existing single argument constructor of GC is probably a good source of inspiration.
The existing GC  constructor could also be adjusted to recommend this method for safety.

What's important is to add the tag `@since 3.126` in the Java-doc in order to make the API-tools aware this method was just added.

Reply:
Thanks for the hint. I hope it is fine with eb1dc57